desertbot/modules/urlfollow/Twitch.py

```diff
@@ -26,61 +41,112 @@ def onLoad(self):
         self.twitchClientID = self.bot.moduleHandler.runActionUntilValue('get-api-key', 'Twitch Client ID')
 
     def follow(self, _: IRCMessage, url: str) -> [str, None]:
-        # Heavily based on Didero's DideRobot code for the same
-        # https://github.com/Didero/DideRobot/blob/06629fc3c8bddf8f729ce2d27742ff999dfdd1f6/commands/urlTitleFinder.py#L37
-        match = re.search(r'twitch\.tv/(?P<twitchChannel>[^/]+)/?(\s|$)', url)
+        match = TWITCH_URL_RE.search(url)
         if not match:
             return
+
         channel = match.group('twitchChannel')
 
-        if self.twitchClientID is None:
-            return '[Twitch Client ID not found]'
-
-        chanData = {}
-        channelOnline = False
-        twitchHeaders = {'Accept': 'application/vnd.twitchtv.v3+json',
-                         'Client-ID': self.twitchClientID}
-        url = 'https://api.twitch.tv/kraken/streams/{}'.format(channel)
-        response = self.bot.moduleHandler.runActionUntilValue('fetch-url', url,
-                                                              extraHeaders=twitchHeaders)
-
-        streamData = response.json()
-
-        if 'stream' in streamData and streamData['stream'] is not None:
-            chanData = streamData['stream']['channel']
-            channelOnline = True
-        elif 'error' not in streamData:
-            url = 'https://api.twitch.tv/kraken/channels/{}'.format(channel)
-            response = self.bot.moduleHandler.runActionUntilValue('fetch-url', url,
-                                                                  extraHeaders=twitchHeaders)
-            chanData = response.json()
-
-        if len(chanData) == 0:
+        #if self.twitchClientID is None:
+        #    return '[Twitch Client ID not found]'
+
+        headers = {
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.150 Safari/537.36",
+            "Accept": "*/*",
```

Comment:
I'm not exactly sure why we need to accept everything here. Would the default accept header in WebUtils not suffice for this?

Reply:
I was sticking very close to the actual calls Twitch is doing, their JS is explicitly setting Accept: */*. To you know, not stick out.

Reply:
Fair enough.